types/encrypted_key.go

```diff
@@ -133,17 +159,31 @@ func (ek *EncryptedKey) DecryptSymmetricKey(cert *tls.Certificate) (cipher.Block
 			}
 
 			return b, nil
-		case MethodRSAPKCS15:
+		case MethodRSAv1_5:
 			pt, err := rsa.DecryptPKCS1v15(rand.Reader, pk, cipherText)
 			if err != nil {
 				return nil, fmt.Errorf("rsa internal error: %v", err)
 			}
 
-			b, err := aes.NewCipher(pt)
+			//From https://docs.oasis-open.org/security/saml/v2.0/saml-core-2.0-os.pdf the xml encryption
+			//methods to be supported are from http://www.w3.org/2001/04/xmlenc#Element.
+			//https://www.w3.org/TR/2002/REC-xmlenc-core-20021210/Overview.html#Element.
+			//https://www.w3.org/TR/2002/REC-xmlenc-core-20021210/#sec-Algorithms
+			//Sec 5.4 Key Transport:
+			//The RSA v1.5 Key Transport algorithm given below are those used in conjunction with TRIPLEDES
+			//Please also see https://www.w3.org/TR/xmlenc-core/#sec-Algorithms and
+			//https://www.w3.org/TR/xmlenc-core/#rsav15note.
+			b, err := des.NewTripleDESCipher(pt)
 			if err != nil {
 				return nil, err
 			}
 
+			// FIXME: The version we had previously in our fork, AES seems more secure from my Googling.
+			// b, err := aes.NewCipher(pt)
+			// if err != nil {
+			// 	return nil, err
+			// }
```

Comment:
@ElizabethStirling This is the only conflict that I can make a confident decision on, do you have an opinion?

FYI, Go tests passes either way.

Reply:
AES sounds much better, let's go for that, if tests still pass with it.

Reply:
Thanks for the input! I'm gonna merge as-is and address this in another commit so we don't lose track in this huge merge commit.

Reply:
https://github.com/sourcegraph/gosaml2/commit/70944041979a86a57c73708d24d4b80b759d14ea